CMakeLists.txt

```
@@ -408,7 +408,7 @@ endif()
 
 if (DEFINED GIT_EXE AND IS_DIRECTORY "${CMAKE_SOURCE_DIR}/enterprise/.git")
```

```suggestion
if (DEFINED GIT_EXE AND USE_ENTERPRISE AND IS_DIRECTORY "${CMAKE_SOURCE_DIR}/enterprise/.git")
```

Must present otherwise cached unrelated `./enterprise` could be used.

---

LGTM